sympy/combinatorics/homomorphisms.py

```diff
@@ -301,8 +301,8 @@ def homomorphism(domain, codomain, gens, images=[], check=True):
     gens = list(gens)
     images = list(images)
 
-    images.extend([codomain.identity]*(len(generators)-len(images)))
-    gens.extend([g for g in generators if g not in gens])
+    images.extend((codomain.identity,) * (len(generators)-len(images)))
+    gens.extend(g for g in generators if g not in gens)
```

Comment:
I'm not sure the second change has a speed improvement

Reply:
Also if `gens` can be a large list then `g not in gens` is an `O(N)` containment test whereas a set could give `O(1)` containment testing. Potentially that makes the difference between the operation as a whole being `O(N)` or `O(N^2)`.

Reply:
You are probably correct. I'll revert those (and see if I find other where a list may help).

Reply:
What about `gens = list(uniq(gens + generators))`?